### DISCUSSION TITLE
Comments on new Assembly.md 2023-01-20

### DISCUSSION THREAD

**Discussion by daveclifford** [DCC-EX/EX-T3-WiFi#40] [General]
Great Job, Alex,  on the new **Assembly.md** update! 

Steps are clearer and addition of your new pictures immensely help in the assembly of the WiFi Throttle. 

My thanks to you for the effort and great thought!  

I believe that this project will be executed by many others looking for a relatively easy project with off-the-shelf components. 

The mechanical aspects are straight forward. 

The code compilation/upload is more daunting! 

But the outcome? Ooh-La-La! A compact, well-engineered and constructed DCC Touch Screen Wireless Throttle for $80 USD (excluding the case 3-D printing)!!! Mine cost me $70 USD from a commercial 3-D printer. 

The only issue I had was to find a source for the M2x5 Self-Tapping screws (odd size) - E-Bay came through - but  a slow boat from China took several weeks to get.

I will keep giving you suggesions to improve the Throttle.

Dave

**Comment by alex-code** [comment]
The screw sizes are a bit specialist but are the biggest I could use within the holes on the PCBs.
For me it's easier to work in Metric but there may be an equivalent size in Imperial which could work.

**Reply by daveclifford** [reply]
DEC.       FRAC       
0.0625 | 1/16   | 
0.0787 | 2 mm | <======== M2  screw 
0.0938 | 3/32   | 

I'll stick with the M2 - I think the others are too small or large

Dave

**Comment by Alan-Lomax** [comment]
Completely agree that the new instructions are clearer and much more straight forward. 
My input from my first buildup that happened today ...
1) Filing and fettling to get the screen to fit was more than a just a little. Got there in the end. I also took a file to the ends of the PCB on the screen (I understand not everyone might be comfortable with this.) We are talking about something like 1/64 from each end.
2) My Screen came with a typical protective film over it and that film had a small green tab. I do recommend keeping the film in place during assembly but be sure to get that tab on the outside as you assemble - else you will have to go back in and fish it out.
3) For the slide switch - the center post (closest to the firebeetle) is a weak link Mine snapped while starting the screw. Ended up using some CA glue
4) My battery had reversed polarity. Note on the firebeetle PCB there is a + and - marking showing what you 'need'. The + is furthest from the USB connector and so this needs to be the RED wire on the battery. The white battery connector has a notch on one side so it will only go on one way. To reverse the wires it is easiest to do this on the battery cable (no soldering or cutting of wires) as follows - with a very small jewelers screwdriver lift up the small tab that retains the wire in the white connector- with the tab up slightly the wire and its metal end should slide out of the connector (no forcing needed). Take note of the orientation so you can slide the other wire in the same way Remove each wire and reinsert them the correct way around. As you slide in the metal end the tab should retain it. (Gentle tug test to verify)
5) At the end of assembly we are just left hanging ... what is next? If I plug it in should something happen? (No but it is not obvious where to carry on). I have VS loaded, and Platform IO so presumably I am to load something on the firebeetle in addition to the files I copied to the micro SD card currently installed on the back of the display! I think a pointer to the next procedure to pick up would be handy.

**Reply by Alan-Lomax** [reply]
One other thing - Before beginning it is necessary to solder the header pins onto the firebeetle (on top of PCB) and also trim off the long leads that stick below the PCB. Do this before mounting it to the frame.

**Reply by alex-code** [reply]
Some more notes I can add to the docs.
Links going backwards and forwards through the stages are a good idea.

There's not much I can do about the 3D parts as I'd imagine each printer would produce something slightly different which does mean a bit of fettling. The type of plastic you use could affect how durable the parts are too, I used PETG.

Do you get the FireBeetle without pre-soldered heads?

**Reply by Alan-Lomax** [reply]
**Re: The links** yes that would be good. I will take a look into the software load process today.

**Agree about 3D printed parts** - just something to be aware of. I printed with PLA. I do have some Carbon Fiber PETG and if the PLA is not up to the task I will do a reprint using that.

**Re: without Pre Soldered leads**
Yes in the box it came with were loose pins. No doubt it was ordered that way. The real point is that if others do the same be sure to solder them in the right way around, and trim the leads form the underside before starting assembly.

**Comment by daveclifford** [comment]
Alex - I would emphasize the* DFR0654-F * (with Headers) version to
purchase (in BIG, BOLD FONTS),

For an in- or  low-experienced assembler, there is much more risk in having
to solder the headers (cold joints, missing connections, shorts, etc!!!)
and creating a $9.00 brick!

[https://www.dfrobot.com/product-2231.html]
FireBeetle 2 ESP32-E IoT Microcontroller* with Header *(Supports Wi-Fi &
Bluetooth)
*SKU:DFR0654-F* $9.90

FireBeetle 2 ESP32-E IoT Microcontroller (Supports Wi-Fi & Bluetooth)
SKU:DFR0654  $8.90

US prices,

Dave Clifford
